controller/src/beerocks/master/db/db.cpp

```diff
@@ -2117,6 +2117,324 @@ bool db::get_hostap_is_acs_enabled(std::string mac)
+               << " scan, setting channel pool succeeded!";
+
+    return true;
+}
+
+const std::set<uint8_t> &db::get_channel_scan_pool(const std::string &mac, const bool single_scan)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return *(std::make_shared<std::set<uint8_t>>());
+    } else if (n->get_type() != beerocks::TYPE_SLAVE || n->hostap == nullptr) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid hostap!";
+        return *(std::make_shared<std::set<uint8_t>>());
+    }
+
+    auto scan_config =
+        (single_scan) ? n->hostap->single_scan_config : n->hostap->continuous_scan_config;
+
+    return scan_config->channel_pool;
+}
+
+bool db::is_channel_in_pool(const std::string &mac, const uint8_t channel, const bool single_scan)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_SLAVE || n->hostap == nullptr) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid hostap!";
+        return false;
+    }
+
+    auto scan_config =
+        (single_scan) ? n->hostap->single_scan_config : n->hostap->continuous_scan_config;
+    auto &channel_pool = scan_config->channel_pool;
+
+    return channel_pool.find(channel) != channel_pool.end();
+}
+
+bool db::clear_channel_scan_results(const std::string &mac, const bool single_scan)
+{
+
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_SLAVE || n->hostap == nullptr) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid hostap!";
+        return false;
+    }
+
+    auto &scan_results =
+        (single_scan) ? n->hostap->single_scan_results : n->hostap->continuous_scan_results;
+
+    scan_results.clear();
+    return true;
+}
+
+bool db::add_channel_scan_results(const std::string &mac,
+                                  const sChannelScanResultsElement &scan_result,
+                                  const bool single_scan)
+{
+
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_SLAVE || n->hostap == nullptr) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid hostap!";
+        return false;
+    }
+
+    auto &scan_results =
+        (single_scan) ? n->hostap->single_scan_results : n->hostap->continuous_scan_results;
+
+    scan_results.push_back(scan_result);
+
+    return true;
+}
+
+const std::list<sChannelScanResultsElement> &db::get_channel_scan_results(const std::string &mac,
+                                                                          const bool single_scan)
+{
+    auto n = get_node(mac);
+    static std::list<sChannelScanResultsElement> dummy_return;
```

Comment:
`return {}` can work as well

Reply:
Tried that, when I do I get an error saying:
> error: returning reference to temporary [-Werror=return-local-addr]

Reply:
> Or you can return a shared_ptr instead of const reference and return nullptr instead of dummy_return.

This will be the only thing that works. But it's super-annoying, because it means the caller has to check for nullptr everywhere while it could just work with the empty set without problem.

Reply:
> > Or you can return a shared_ptr instead of const reference and return nullptr instead of dummy_return.
> 
> This will be the only thing that works. But it's super-annoying, because it means the caller has to check for nullptr everywhere while it could just work with the empty set without problem.

So you think the `dummy_return` should stay?

Reply:
for now it stays, if we decide to fix this later on we'll decide in #665